compiler/rustc_interface/src/queries.rs

```diff
@@ -324,7 +323,7 @@ impl Compiler {
                 .time("serialize_dep_graph", || gcx.enter(rustc_incremental::save_dep_graph));
         }
 
-        _timer = Some(self.session().timer("free_global_ctxt"));
+        let _timer = self.session().timer("free_global_ctxt");
```

Comment:
The timer will be dropped before Queries and thus GlobalCtxt this way.

Reply:
True. I'll switch to an explicit `drop`.

Reply:
I couldn't get that to work. I've gone back to the original code but added some comments.